ja2-stracciatella/ja2-stracciatella
# Externalizing Movement Costs

Externalizing Movement Costs on the strategic map (Strategic_Movement_Costs.cc). 

Again, see #665 and #1011.

This PR conflicts with #1018 and #1020. This needs to be re-based and re-tested before merging to master.

## Summary

- JSON data is a direct translation from the original C++ code
- Extracted tables replaced the arrays: `g_traverse_we`, `g_traverse_ns`, `g_traverse_through` and `g_travel_rating`
   - Noted that `ubTravelRating` is not used for anything meaningful
- Removed a unit test which tests on the table data; Added equivalent Assert statements in model class.

## Testing

Manually tested like this:

1. Start a new game, with 0.16.1
2. Hire somebody and proceed to land in A9; GABBI-killed all enemies in A9
3. Try to move across strategic map. Record the ETA to different destinations.
4. Repeat 1~3 after applying this PR to master. Verify ETAs are exactly the same
5. Apply the data file override for Wildfire Maps mod, which modifies some traversibility values. Repeat 1~3 and verify getting different ETAs.

Expected results:
| From | To | Travel ETA  (0.16.1) |
|---|---|---|
| A9 | A10 | 5m |
| A9 | A8 | 1h 44m |
| A9 | A1| 15h 40m |
| A9 | A15 | 10h 13m |
| A9 | I6 | 12h 52m|
| A9 | H1| 15h 30m |
| A9 | I16 | 15h 45m |
| A9 | P1 | 22h 6m |
| A9 | K4 | 22h 32m |
| A9 | M13 | 20h 31m |
